tests/unit/workflows/test_workflows_tasks_actions.py

```diff
@@ -398,6 +399,48 @@ def test_is_submission_returns_false_if_obj_has_falsy_acquisition_source():
     assert not is_submission(obj, eng)
 
 
+def test_fix_submission_number():
+    schema = load_schema('hep')
+    subschema = schema['properties']['acquisition_source']
+
+    data = {
+        'acquisition_source': {
+            'method': 'hepcrawl',
+            'submission_number': '751e374a017311e896d6fa163ec92c6a',
+        },
+    }
+    extra_data = {}
+    assert validate(data['acquisition_source'], subschema) is None
+
+    obj = MockObj(data, extra_data)
+    eng = MockEng()
+
+    fix_submission_number(obj, eng)
+
+    assert obj.data['acquisition_source']['submission_number'] == 1
```

Comment:
Mh, I missed the obvious bug here. Can you guess which one, @iulianav, and how we could have prevented it?